rest/client/request.go

```diff
@@ -49,6 +46,16 @@ const (
 	patch         = "PATCH"
 )
 
+type HTTPConflictError struct{}
+
+func (e HTTPConflictError) Error() string {
+	return "Received status code 409"
+}
+
+func (e HTTPConflictError) Cause() error {
+	return e
+}
```

Comment:
This is actually pretty clever, and I think might work, but doesn't quite need to happen

you could just: 

var HTTPConflictError = errors.New(evergreen.TaskConflict)

and then you don't need to instnatiate the type (even though that's not really expensive) at the call site. errors.Cause() returns the root error object, which means you can wrap with impunity.

Reply:
Fixed.